src/codegen/mod.rs

```diff
@@ -287,7 +287,9 @@ impl CodeGenerator for Item {
         }
 
         debug!("<Item as CodeGenerator>::codegen: self = {:?}", self);
-        assert!(whitelisted_items.contains(&self.id()));
+        if !whitelisted_items.contains(&self.id()) {
+            error!("Found non-whitelisted item in code generation: {:?}", self);
```

Comment:
Please add `TODO #XXX: we should turn this back into an assertion` after you've filed the follow up bug. Thanks!